crates/templates/src/run.rs

```diff
@@ -429,29 +431,79 @@ impl Run {
     }
 
     fn partials(&self) -> anyhow::Result<impl liquid::partials::PartialCompiler> {
-        let mut partials = PartialsBuilder::empty();
-
-        if let Some(partials_dir) = self.template.partials_dir() {
-            let partials_dir = std::fs::read_dir(partials_dir)
-                .context("Error opening template partials directory")?;
-            for partial_file in partials_dir {
-                let partial_file =
-                    partial_file.context("Error scanning template partials directory")?;
-                if !partial_file.file_type().is_ok_and(|t| t.is_file()) {
-                    anyhow::bail!("Non-file in partials directory: {partial_file:?}");
+        match self.template.partials_dir() {
```

Comment:
It's not clear to me why partials support needs to change as part of this.  Should this be a separate PR or is it needed here?